tests/providers/node/pool_test.py

```diff
@@ -0,0 +1,47 @@
+import pytest
+
+from ethtx.providers.node.base import NodeBase
+from ethtx.providers.node.pool import NodeConnectionPool
+
+MAINNET_CHAIN = {"mainnet": {"hook": "a", "poa": True}}
+GOERLI_CHAIN = {"goerli": {"hook": "a, b, c", "poa": False}}
+
+GOERLI_NODE = NodeBase("B", False)
+
+
+class TestNodeConnectionPool:
+    @classmethod
+    def setup_class(cls):
+        cls.pool = NodeConnectionPool(nodes=MAINNET_CHAIN)
+
+    def test_one_connections(self):
+        data = NodeBase(node="a", poa=True)
+        assert self.pool.connections
+        assert len(self.pool) == 1
+        assert self.pool.connections["mainnet"][0].__dict__ == dict(data)
+        assert "mainnet" in self.pool.connections
+
+    def test_ok_getattr(self):
+        assert self.pool.mainnet
+
+    def test_wrong_getattr(self):
+        with pytest.raises(AttributeError):
+            _ = self.pool.goerli
+
+    def test_setattr(self):
+        self.pool.goerli = GOERLI_NODE
+        assert self.pool.goerli
+
+    def test_wrong_setattr(self):
+        with pytest.raises(ValueError) as e:
+            self.pool.wrong = 1, 1, 1
+
+        assert str(e.value) == "Value is not instance of NodeBase"
+
+    def test_multiple_connections(self):
+        assert self.pool.connections
+        assert len(self.pool) == 2
+        assert "mainnet" in self.pool.connections
+        assert "goerli" in self.pool.connections
+        assert self.pool.mainnet
+        assert self.pool.goerli
```

Comment:
1. This whole thing should be different after change to NodeConnectionPool
2. Names, those are names that do not say what is tested and are meaningless
3. Asserts are missing error messages
4. Single responsibility as before

Reply:
Done